cartographer/io/fake_file_writer_test.cc

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <vector>
+
+#include "cartographer/io/fake_file_writer.h"
+#include "gtest/gtest.h"
+
+namespace cartographer {
+namespace io {
+namespace {
+
+TEST(FakeFileWriter, Write) {
+  const std::string header = "dummy header";
+  const std::vector<std::string> data_stream = {"data 1", "data 2"};
+  std::stringstream expected_result;
+
+  FakeFileWriter writer;
+  writer.WriteHeader(header.c_str(), header.size());
```

Comment:
It's better to check different things in separate tests, i.e. have separate tests for `Write` and `WriteHeader`.